swarm/pss/pss.go

```diff
@@ -529,6 +531,9 @@ func (p *Pss) isSelfPossibleRecipient(msg *PssMsg, prox bool) bool {
 // The value in `address` will be used as a routing hint for the
 // public key / topic association
 func (p *Pss) SetPeerPublicKey(pubkey *ecdsa.PublicKey, topic Topic, address *PssAddress) error {
+	if !checkAddress(address) {
+		return errors.New("invalid address")
```

Comment:
Probably a good idea to export this to `ErrInvalidAddress` in case clients want to handle it.

Reply:
Agree about the errors, and reading through that should change for most errors in that case. I'll make a separate PR about this.